core/src/main/java/org/infinispan/stats/impl/ClusterCacheStatsImpl.java

```diff
@@ -142,11 +141,16 @@ void updateStats() {
       putIntAttributes(responseList, NUMBER_OF_LOCKS_AVAILABLE);
       putIntAttributesMax(responseList, REQUIRED_MIN_NODES);
 
-      long numberOfEntriesInMemory = getCacheMode(cache).isReplicated() ?
-            cache.getStats().getCurrentNumberOfEntriesInMemory() :
-            (long) addDoubleAttributes(responseList, NUMBER_OF_ENTRIES_IN_MEMORY);
-      statsMap.put(NUMBER_OF_ENTRIES_IN_MEMORY, numberOfEntriesInMemory);
-      statsMap.put(NUMBER_OF_ENTRIES, cache.size());
+      if (accurateSize) {
+         // Count each entry only once
+         long numberOfEntriesInMemory = cache.withFlags(Flag.SKIP_CACHE_LOAD).size();
+         statsMap.put(NUMBER_OF_ENTRIES_IN_MEMORY, numberOfEntriesInMemory);
+         int numberOfEntries = cache.size();
+         statsMap.put(NUMBER_OF_ENTRIES, (long) numberOfEntries);
+      } else {
+         statsMap.put(NUMBER_OF_ENTRIES_IN_MEMORY, -1L);
+         statsMap.put(NUMBER_OF_ENTRIES, -1L);
```

Comment:
@karesti this should "fix" the console, as in the size will be -1 instead of 0 when accurate-size="false", so the user will not think the cache is empty. Of course, the console could replace -1 with "N/A" with a tooltip telling users to enable accurate-size to make it even clearer :D

If you want, I could also change line to always pass `accurateSize=true` to the `DistributedCacheStatsCallable` constructor -- then the size would always be computed, at least until the approximate size changes are done.